bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/composite/AbstractCompositeModuleHandler.java

```diff
@@ -88,10 +89,14 @@ public AbstractCompositeModuleHandler(M module, MT moduleType, LinkedHashMap<M,
 
     @Override
     public void dispose() {
-        if (moduleHandlerMap != null) {
-            moduleHandlerMap.clear();
-            moduleHandlerMap = null;
+        List<M> children = getChildren();
+        for (M child : children) {
+            ModuleHandler childHandler = moduleHandlerMap.remove(child);
+            childHandler.dispose();
         }
```

Comment:
Wouldn't it be safer to additionally still call 
```
handler.setRuleEngineCallback(null);
```
afterwards, in case a handler "forgets" to dereference the callback in a dispose?
If we translate "dispose->deactivate" and "setRuleEngineCallback(null)->unsetRuleEngineCallback()" in a DS component, it would happen in a similar way: The reference that is injected from externally would also be revoked from externally.
I e.g. see that GenericEventTriggerHandler is NOT releasing the callback.